### PR TITLE
[enhancement] Container delete confirmation

### PR DESCRIPTION
**Pull request by bakbakbakbakbak**
Added a confirmation dialog that proceeds with deleting a container if user selects "Ok".

This only shows when deleting a container that is containing items.

### CODE REVIEW

**Comment by bakbakbakbakbak** [comment]
resolves #298, resolves #295

File: src/templates/items/delete-container.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<form class="ose dialog">
+    <div>
+        {{localize 'OSE.dialog.confirmDeleteContainer'}}
+        Are you sure you want to delete the container?
```
Comment (wyrmisis) [inline src/templates/items/delete-container.html:4]: Should this be localized?

File: src/module/actor/actor-sheet.js [hunk 2]
```diff
@@ -533,8 +533,39 @@ export class OseActorSheet extends ActorSheet {
...
+
+      if (item.type === "container" && itemData.itemIds.length > 0) {
+        // Create dialog
+        let templateData = { types:["ok", "cancel"] },
+          dlg = await renderTemplate(
+            `${OSE.systemPath()}/templates/items/delete-container.html`,
+            templateData
+          );
+  
+        const confirmation = new Dialog({
```
Comment (wyrmisis) [inline src/module/actor/actor-sheet.js:548]: You should be able to use `Dialog.confirm` for this: https://foundryvtt.com/api/v10/classes/client.Dialog.html#confirm

**Review by wyrmisis — CHANGES_REQUESTED** [review]
Solid changes, I left some thoughts.

File: src/templates/items/delete-container.html (hunk 1 above)
Comment (bakbakbakbakbak) [reply src/templates/items/delete-container.html]: You're right! I missed to remove my non-localized text!

File: src/module/actor/actor-sheet.js (hunk 2 above)
Comment (bakbakbakbakbak) [reply src/module/actor/actor-sheet.js]: So much more elegant, thanks, I have now changed to using `Dialog.config` instead.

File: src/module/actor/actor-sheet.js [hunk 3]
```diff
@@ -533,8 +533,20 @@ export class OseActorSheet extends ActorSheet {
       item.sheet.render(true);
     });
 
-    html.find(".item-delete").click((event) => {
-      this._removeItemFromActor(event);
+    html.find(".item-delete").click(async (event) => {
```
Comment (wyrmisis) [inline src/module/actor/actor-sheet.js:536]: This arrow function is async, but we're not awaiting anywhere. Should we be awaiting something?

Comment (bakbakbakbakbak) [reply src/module/actor/actor-sheet.js]: No, I missed that when I replaced the custom dialog with the Dialog.confirm - cleaned up in latest commit!

File: src/module/actor/actor-sheet.js [hunk 4]
```diff
@@ -534,7 +534,19 @@ export class OseActorSheet extends ActorSheet {
     });
 
     html.find(".item-delete").click((event) => {
-      this._removeItemFromActor(event);
+      const item = this._getItemFromActor(event);
+      const itemData = item?.system;
+
+      if (item.type === "container" && itemData.itemIds.length > 0) {
```
Comment (wyrmisis) [inline src/module/actor/actor-sheet.js:540]: This is more personal preference than anything, but I'm generally a fan of as few branches as possible:

```js
if (item.type !== 'container' || !item?.system?.itemIds?.length)
  return this._removeItemFromActor(event);
  
Dialog.confirm({...});  
```

Comment (wyrmisis) [inline src/module/actor/actor-sheet.js:540]: `itemdata = item?.system` is something we'd do if we expect to get here without having a valid item to delete. If so, we might want to add similar checks to the `if` statement here.

**Review by wyrmisis — CHANGES_REQUESTED** [review]
Sorry, just one more change (and a suggestion)

File: src/module/actor/actor-sheet.js (hunk 4 above)
Comment (bakbakbakbakbak) [reply src/module/actor/actor-sheet.js:540]: Yes, I realize that I declared `itemData` only as a helper in the if-statement, so it makes no sense to keep given your suggested change above  - fixed in latest commit!

Comment (bakbakbakbakbak) [reply src/module/actor/actor-sheet.js:540]: I agree - fixed in latest commit!

File: src/module/actor/actor-sheet.js [hunk 5]
```diff
@@ -534,7 +534,17 @@ export class OseActorSheet extends ActorSheet {
     });
 
     html.find(".item-delete").click((event) => {
-      this._removeItemFromActor(event);
+      const item = this._getItemFromActor(event);
+
+      if (item.type !== "container" || !item?.system?.itemIds?.length > 0)
```
Comment (wyrmisis) [inline src/module/actor/actor-sheet.js:539]: `item.type` => `item?.type`, since we're checking on the second condition

**Review by wyrmisis — CHANGES_REQUESTED** [review]
Just one last little change!

File: src/module/actor/actor-sheet.js (hunk 5 above)
Comment (bakbakbakbakbak) [reply src/module/actor/actor-sheet.js]: Indeed, this makes sense!

**Review by wyrmisis — APPROVED** [review]
Looks good, thanks for your patience!

**Comment by anthonyronda** [comment]
I was able to test the change in the game system and I'm satisfied it works as intended. Thanks!

**Pull request merged**